### ISSUE TITLE
EasyOCR (1.7.0: `-215:Assertion failed) !ssize.empty()` cv2 Error

### ISSUE THREAD

**Issue by Casyfill** [JaidedAI/EasyOCR#1034]
Versions:
- EasyOCR 1.7.0
- OpenCV 4.7.0

Code (jupyter):
```python
READER = easyocr.Reader(['en'])

path = './1470646.jpg'
result = READER.readtext(path)
```
Results in this Error:
```
File ~/Library/Caches/pypoetry/virtualenvs/floorplans-ocr-TIa0fr6E-py3.8/lib/python3.8/site-packages/easyocr/easyocr.py:383, in Reader.recognize(self, img_cv_grey, horizontal_list, free_list, decoder, beamWidth, batch_size, workers, allowlist, blocklist, detail, rotation_info, paragraph, contrast_ths, adjust_contrast, filter_ths, y_ths, x_ths, reformat, output_format)
    381 h_list = [bbox]
    382 f_list = []
--> 383 image_list, max_width = get_image_list(h_list, f_list, img_cv_grey, model_height = imgH)
    384 result0 = get_text(self.character, imgH, int(max_width), self.recognizer, self.converter, image_list,\
    385               ignore_char, decoder, beamWidth, batch_size, contrast_ths, adjust_contrast, filter_ths,\
    386               workers, self.device)
    387 result += result0

File ~/Library/Caches/pypoetry/virtualenvs/floorplans-ocr-TIa0fr6E-py3.8/lib/python3.8/site-packages/easyocr/utils.py:613, in get_image_list(horizontal_list, free_list, img, model_height, sort_output)
    611     pass
    612 else:
--> 613     crop_img,ratio = compute_ratio_and_resize(crop_img,width,height,model_height)
    614     image_list.append( ( [[x_min,y_min],[x_max,y_min],[x_max,y_max],[x_min,y_max]] ,crop_img) )
    615     max_ratio_hori = max(ratio, max_ratio_hori)

File ~/Library/Caches/pypoetry/virtualenvs/floorplans-ocr-TIa0fr6E-py3.8/lib/python3.8/site-packages/easyocr/utils.py:574, in compute_ratio_and_resize(img, width, height, model_height)
    572 if ratio<1.0:
    573     ratio = calculate_ratio(width,height)
--> 574     img = cv2.resize(img,(model_height,int(model_height*ratio)), interpolation=Image.ANTIALIAS)
    575 else:
    576     img = cv2.resize(img,(int(model_height*ratio),model_height),interpolation=Image.ANTIALIAS)

error: OpenCV(4.7.0) /Users/runner/work/opencv-python/opencv-python/opencv/modules/imgproc/src/resize.cpp:4062: error: (-215:Assertion failed) !ssize.empty() in function 'resize'
```

Image itself is fine and both PIL and cv2 can read it and shows correct img.shape

**Comment by Casyfill** [comment]
here is the image example (albeit I have similar issue with other images in the batch):
[link](https://www.dropbox.com/s/jasag24sniqalg2/1272917.jpg?dl=0)

**Comment by specialprocedures** [comment]
I'm also experiencing this issue, both on 1.6.2 and 1.7.0.
`
---------------------------------------------------------------------------
error                                     Traceback (most recent call last)
File [~/miniconda3/lib/python3.10/site-packages/easyocr/easyocr.py:456](https://file+.vscode-resource.vscode-cdn.net/home/ian/dev/personal/~/miniconda3/lib/python3.10/site-packages/easyocr/easyocr.py:456), in Reader.readtext(self, image, decoder, beamWidth, batch_size, workers, allowlist, blocklist, detail, rotation_info, paragraph, min_size, contrast_ths, adjust_contrast, filter_ths, text_threshold, low_text, link_threshold, canvas_size, mag_ratio, slope_ths, ycenter_ths, height_ths, width_ths, y_ths, x_ths, add_margin, threshold, bbox_min_score, bbox_min_size, max_candidates, output_format)
    [454](file:///home/ian/miniconda3/lib/python3.10/site-packages/easyocr/easyocr.py?line=453) # get the 1st result from hor & free list as self.detect returns a list of depth 3
    [455](file:///home/ian/miniconda3/lib/python3.10/site-packages/easyocr/easyocr.py?line=454) horizontal_list, free_list = horizontal_list[0], free_list[0]
--> [456](file:///home/ian/miniconda3/lib/python3.10/site-packages/easyocr/easyocr.py?line=455) result = self.recognize(img_cv_grey, horizontal_list, free_list,\
    [457](file:///home/ian/miniconda3/lib/python3.10/site-packages/easyocr/easyocr.py?line=456)                         decoder, beamWidth, batch_size,\
    [458](file:///home/ian/miniconda3/lib/python3.10/site-packages/easyocr/easyocr.py?line=457)                         workers, allowlist, blocklist, detail, rotation_info,\
    [459](file:///home/ian/miniconda3/lib/python3.10/site-packages/easyocr/easyocr.py?line=458)                         paragraph, contrast_ths, adjust_contrast,\
    [460](file:///home/ian/miniconda3/lib/python3.10/site-packages/easyocr/easyocr.py?line=459)                         filter_ths, y_ths, x_ths, False, output_format)
    [462](file:///home/ian/miniconda3/lib/python3.10/site-packages/easyocr/easyocr.py?line=461) return result

File [~/miniconda3/lib/python3.10/site-packages/easyocr/easyocr.py:379](https://file+.vscode-resource.vscode-cdn.net/home/ian/dev/personal/~/miniconda3/lib/python3.10/site-packages/easyocr/easyocr.py:379), in Reader.recognize(self, img_cv_grey, horizontal_list, free_list, decoder, beamWidth, batch_size, workers, allowlist, blocklist, detail, rotation_info, paragraph, contrast_ths, adjust_contrast, filter_ths, y_ths, x_ths, reformat, output_format)
    [377](file:///home/ian/miniconda3/lib/python3.10/site-packages/easyocr/easyocr.py?line=376) h_list = [bbox]
    [378](file:///home/ian/miniconda3/lib/python3.10/site-packages/easyocr/easyocr.py?line=377) f_list = []
--> [379](file:///home/ian/miniconda3/lib/python3.10/site-packages/easyocr/easyocr.py?line=378) image_list, max_width = get_image_list(h_list, f_list, img_cv_grey, model_height = imgH)
    [380](file:///home/ian/miniconda3/lib/python3.10/site-packages/easyocr/easyocr.py?line=379) result0 = get_text(self.character, imgH, int(max_width), self.recognizer, self.converter, image_list,\
    [381](file:///home/ian/miniconda3/lib/python3.10/site-packages/easyocr/easyocr.py?line=380)               ignore_char, decoder, beamWidth, batch_size, contrast_ths, adjust_contrast, filter_ths,\
    [382](file:///home/ian/miniconda3/lib/python3.10/site-packages/easyocr/easyocr.py?line=381)               workers, self.device)
    [383](file:///home/ian/miniconda3/lib/python3.10/site-packages/easyocr/easyocr.py?line=382) result += result0

File [~/miniconda3/lib/python3.10/site-packages/easyocr/utils.py:571](https://file+.vscode-resource.vscode-cdn.net/home/ian/dev/personal/~/miniconda3/lib/python3.10/site-packages/easyocr/utils.py:571), in get_image_list(horizontal_list, free_list, img, model_height, sort_output)
    [569](file:///home/ian/miniconda3/lib/python3.10/site-packages/easyocr/utils.py?line=568)     pass
    [570](file:///home/ian/miniconda3/lib/python3.10/site-packages/easyocr/utils.py?line=569) else:
--> [571](file:///home/ian/miniconda3/lib/python3.10/site-packages/easyocr/utils.py?line=570)     crop_img,ratio = compute_ratio_and_resize(crop_img,width,height,model_height)
    [572](file:///home/ian/miniconda3/lib/python3.10/site-packages/easyocr/utils.py?line=571)     image_list.append( ( [[x_min,y_min],[x_max,y_min],[x_max,y_max],[x_min,y_max]] ,crop_img) )
    [573](file:///home/ian/miniconda3/lib/python3.10/site-packages/easyocr/utils.py?line=572)     max_ratio_hori = max(ratio, max_ratio_hori)

File [~/miniconda3/lib/python3.10/site-packages/easyocr/utils.py:532](https://file+.vscode-resource.vscode-cdn.net/home/ian/dev/personal/~/miniconda3/lib/python3.10/site-packages/easyocr/utils.py:532), in compute_ratio_and_resize(img, width, height, model_height)
    [530](file:///home/ian/miniconda3/lib/python3.10/site-packages/easyocr/utils.py?line=529) if ratio<1.0:
    [531](file:///home/ian/miniconda3/lib/python3.10/site-packages/easyocr/utils.py?line=530)     ratio = calculate_ratio(width,height)
--> [532](file:///home/ian/miniconda3/lib/python3.10/site-packages/easyocr/utils.py?line=531)     img = cv2.resize(img,(model_height,int(model_height*ratio)), interpolation=Image.ANTIALIAS)
    [533](file:///home/ian/miniconda3/lib/python3.10/site-packages/easyocr/utils.py?line=532) else:
    [534](file:///home/ian/miniconda3/lib/python3.10/site-packages/easyocr/utils.py?line=533)     img = cv2.resize(img,(int(model_height*ratio),model_height),interpolation=Image.ANTIALIAS)

error: OpenCV(4.5.4) [/tmp/pip-req-build-4x5kub8r/opencv/modules/imgproc/src/resize.cpp:4051](https://file+.vscode-resource.vscode-cdn.net/tmp/pip-req-build-4x5kub8r/opencv/modules/imgproc/src/resize.cpp:4051): error: (-215:Assertion failed) !ssize.empty() in function 'resize'`

**Comment by BeastLe9enD** [comment]
same here. I just installed `pip install easyocr` and then tried this simple example:
```Python
import easyocr

def recognize(path):
    reader = easyocr.Reader(['en']);
    result = reader.readtext('example.jpg')
    print(result)
```

Would be nice to have some help here.

**Comment by DivyaPalanisamy001** [comment]
EasyOCR version : 1.6.2
 We are calling easyocr library from python file, passing ndarray as argument to readtext() of easyocr library.

And, its failing with below error, 

detection_result = self.easyocr_reader.readtext(I)<EOL>  File "C:\Python38\lib\site-packages\easyocr\easyocr.py", line 458, in readtext<EOL>    result = self.recognize(img_cv_grey, horizontal_list, free_list,\<EOL>  File "C:\Python38\lib\site-packages\easyocr\easyocr.py", line 379, in recognize<EOL>    image_list, max_width = get_image_list(h_list, f_list, img_cv_grey, model_height = imgH)<EOL>  File "C:\Python38\lib\site-packages\easyocr\utils.py", line 571, in get_image_list<EOL>    crop_img,ratio = compute_ratio_and_resize(crop_img,width,height,model_height)<EOL>  File "C:\Python38\lib\site-packages\easyocr\utils.py", line 532, in compute_ratio_and_resize<EOL>    img = cv2.resize(img,(model_height,int(model_height*ratio)), interpolation=Image.ANTIALIAS)<EOL>****cv2.error: OpenCV(4.5.4) D:\a\opencv-python\opencv-python\opencv\modules\imgproc\src\resize.cpp:4051: error: (-215:Assertion failed) !ssize.empty() in function 'cv::resize'<EOL><EOL>"**** 

During debug, could see that

**Comment by ozyhuboi** [comment]
I noticed that in utils.py, the 'calculate_ratio' function sometimes delivers a negative ratio when the 'img' parameter is an empty ndarray (like '[]'). I think I can just set the ratio to 1 or something when it dips below 0, but I'm not sure the intended behavior here.

**Comment by ozyhuboi** [comment]
![image](https://github.com/JaidedAI/EasyOCR/assets/7039525/8882b345-6561-4c5c-99b7-0a97ab6e46c4)

**Comment by ozyhuboi** [comment]
And this is possibly related to this error from openCV: 
https://stackoverflow.com/questions/31996367/opencv-resize-fails-on-large-image-with-error-215-ssize-area-0-in-funct

**Comment by ozyhuboi** [comment]
TLDR: it seems that in openCV 'modules\imgproc\src\imgwarp.cpp' 
There are two lines causing problems: 
CV_Assert( ssize.area() > 0 );
CV_Assert( dsize.area() > 0 );

Both of which must be commented out.

**Comment by ozyhuboi** [comment]
Or you could actually be processing a null img, which you shouldn't do

**Comment by DivyaPalanisamy001** [comment]
Hi @ozyhuboi , Thank you for the response.

In my case, my input dicom image is 16-bit signed integer(Pixel Representation = 1 and bits Allocated = 16), when it is directly passed to easyocr, it is failing to handle signed int pixel data.

Hence, crop_img is prepared based on horizontal list which is sent as arg to get_image_list(), and this crop_img is sent as empty array to cv2.resize() method of opencv python package, which is actually causing this error. [Attached the screenshot of how crop_img is populated]

So, I tried to convert the 16bit signed int to 8bit unsigned int and passed the image to easyocr. Now able to get the detection results.

Is there any limitation that easyocr cannot process 16bit signed int type pixel data?

Please clarify

**Comment by DivyaPalanisamy001** [comment]
Right now, no issues with the conversion, 

We are trying to do something like below,     
        
if(bits_allocated == 16):
                I = (I - np.min(I)) / (np.max(I) - np.min(I)) * 255
                I = I.astype(np.uint8)

After conversion, we are passing it to easyocr, 
easyocr_reader.readtext(I) to get the detection results.

In this way, its working fine for 16bit dicom images as well.

**Comment by bszollosinagy** [comment]
HI everyone,

I also received the same error, and for me the following was the cause:

- The image was taken with a mobile phone, held in hand vertically, but when the phone saved the JPG, it only indicated the correct orientation in the EXIF metadata.. the image itself was not actually rotated. So if I simply loaded the file into Gimp (the free image editor) then it automatically offered to rotate the image based on the EXIF data. After saving it, the EasyOCR worked as expected.

I hope it helps.

**Comment by coderlegion419** [comment]
The OpenCV error "(-215: Assertion failed) !ssize.empty() in function cv::resize" occurs when the input image to cv::resize is empty or invalid, often due to incorrect file paths, corrupted files, or insufficient permissions. To resolve this, always verify that the image is correctly loaded using cv2.imread() before resizing. Implementing checks like if image is not None can prevent errors and ensure smooth image processing. For a detailed guide on diagnosing and fixing this issue, visit https://coderlegion.com/64/error-215-assertion-failed-ssize-empty-in-function-cv-resize

**Comment by harningle** [comment]
I got the same error, but only if I use small `low_text` and `text_threshold`. The image is attached below.

```python-console
>>> import easyocr
>>> reader = easyocr.Reader(lang_list=['en'])
>>> reader.readtext('image.png')
[]  # Correct
>>> reader.readtext('image.png', low_text=0.1, text_threshold=0.1)
cv2.error: OpenCV(4.12.0) D:\a\opencv-python\opencv-python\opencv\modules\imgproc\src\resize.cpp:4208: error: (-215:Assertion failed) !ssize.empty() in function 'cv::resize'
```

<img width="143" height="102" alt="Image" src="https://github.com/user-attachments/assets/bc297ae7-6e55-4381-a81a-0b8ca6464264" />
